src/extlibs/hrvo/hrvo_agent.cpp

```diff
@@ -34,138 +34,199 @@
+    }
 
-VelocityObstacle HRVOAgent::createVelocityObstacle(const Agent &other_agent)
-{
-    if ((position_ - other_agent.getPosition()).lengthSquared() >
-        std::pow(radius_ + other_agent.getRadius(), 2))
+    // Create Velocity Obstacles for nearby static obstacles
+    Point agent_position_point(getPosition());
+    Circle circle_rep_of_agent(agent_position_point, radius_);
+    Segment path(agent_position_point, Point(current_destination));
+    for (const auto &obstacle : static_obstacles)
     {
-        // This Agent is not colliding with other agent
-        const float angle =
-            (position_ - other_agent.getPosition()).orientation().toRadians();
-
-        // The opening angle of the velocity obstacle
-        // opening angle = arcsin((rad_A + rad_B) / distance_BA)
-        const float openingAngle =
-            std::asin((radius_ + other_agent.getRadius()) /
-                      (position_ - other_agent.getPosition()).length());
-        // Direction of the two edges of the velocity obstacles
-        Vector right_side =
-            Vector::createFromAngle(Angle::fromRadians(angle - openingAngle));
-        Vector left_side =
-            Vector::createFromAngle(Angle::fromRadians(angle + openingAngle));
-
-        const float d = std::sin(2.f * openingAngle);
-
-        // This shifts one side of the velocity obstacle to share the responsibility
-        // of avoiding collision with other agent. This assumes that other agent will also
-        // be running HRVO
-        if ((position_ - other_agent.getPosition())
-                .determinant(other_agent.getPrefVelocity() - pref_velocity_) > 0.0f)
+        double dist_agent_to_obstacle = obstacle->distance(agent_position_point);
+
+        // Set of heuristics to minimize the amount of velocity obstacles
+        if ((obstacle->intersects(path) ||
+             dist_agent_to_obstacle < 2 * ROBOT_MAX_RADIUS_METERS) &&
+            !obstacle->contains(agent_position_point))
         {
-            // Relative velocity is in the right half of velocity obstacle (VO)
-            // Shift the VO apex to the left so the right side is smaller, making the
-            // VO a Hybrid Reciprocal Velocity Obstacle (HRVO)
-            const float s =
-                0.5f * (other_agent.getVelocity() - velocity_).determinant(left_side) / d;
-
-            Vector apex =
-                velocity_ + s * right_side -
-                (position_ - other_agent.getPosition())
-                    .normalize((uncertaintyOffset_ *
-                                (position_ - other_agent.getPosition()).length() /
-                                (radius_ + other_agent.getRadius())));
-            return VelocityObstacle(apex, right_side, left_side);
+            VelocityObstacle velocity_obstacle =
+                obstacle->generateVelocityObstacle(circle_rep_of_agent, Vector());
+            velocityObstacles_.push_back(velocity_obstacle);
```

Comment:
how does this represent a static obstacle?

Reply:
Discussed offline